src/Products/PageTemplates/PageTemplateFile.py

```diff
@@ -178,22 +178,30 @@ def _prepare_html(self, text):
         text = text.decode(encoding)
         return text, type_
 
+    def _prepare_xml(self, text):
+        try:
+            text = text.decode(encodingFromXMLPreamble(text))
+        except UnicodeDecodeError:
+            pass
```

Comment:
+1

Reply:
If I remove the try/except the tests from `test_sniffer_xml_utf16_be` and `test_sniffer_xml_utf16_le` will fail because the regexp in `encodingFromXMLPreamble` method will fail to match the encoding.